common/src/main/scala/net/psforever/packet/game/SquadDefinitionActionMessage.scala

```diff
@@ -84,309 +414,65 @@ final case class SquadDefinitionActionMessage(action : Int,
-    uint4L ::
-      lcodec
-    ).exmap[allPattern] (
-    {
-      case a :: b :: HNil =>
-        Attempt.successful(None :: Some(a) :: None :: Some(b) :: None :: None :: HNil)
-    },
-    {
-      case _ :: None :: _ :: _ :: _ :: _ :: HNil =>
-        Attempt.failure(Err("expected a integer value but found nothing"))
-      case _ :: _ :: _ :: None :: _ :: _ :: HNil =>
-        Attempt.failure(Err("expected a long value but found nothing"))
-      case _ :: Some(a) :: _ :: Some(b) :: _ :: _ :: HNil =>
-        Attempt.successful(a :: b :: HNil)
-    }
-  )
-
-  /**
-    * `Codec` for reading an `Int` followed by a `String` from remaining stream data.
-    * All `String`s processed by this reading are wide character and are padded by two.
-    * @return a filled-out `allPattern` if successful
-    */
-  def intStringCodec : Codec[allPattern] = (
-    uint4L ::
-      PacketHelpers.encodedWideStringAligned(2)
-    ).exmap[allPattern] (
-    {
-      case a :: b :: HNil =>
-        Attempt.successful(Some(b) :: Some(a) :: None :: None :: None :: None :: HNil)
-    },
-    {
-      case None:: _ :: _ :: _ :: _ :: _ :: HNil =>
-        Attempt.failure(Err("expected a string value but found nothing"))
-      case _ :: None :: _ :: _ :: _ :: _ :: HNil =>
-        Attempt.failure(Err("expected an integer value but found nothing"))
-      case Some(b) :: Some(a) :: _ :: _ :: _ :: _ :: HNil =>
-        Attempt.successful(a :: b :: HNil)
-    }
-  )
+  def selectFromActionCode(code : Int) : Codec[SquadAction] = {
+    import SquadAction.Codecs._
+    import scala.annotation.switch
+    ((code : @switch) match {
+      case 0 => displaySquadCodec
+      case 1 => squadMemberInitializationIssueCodec
+      case 3 => saveSquadFavoriteCodec
+      case 4 => loadSquadFavoriteCodec
+      case 5 => deleteSquadFavoriteCodec
+      case 7 => listSquadFavoriteCodec
+      case 8 => requestListSquadCodec
+      case 9 => stopListSquadCodec
+      case 10 => selectRoleForYourselfCodec
+      case 15 => cancelSelectRoleForYourselfCodec
+      case 16 => associateWithSquadCodec
+      case 17 => setListSquadCodec
+      case 19 => changeSquadPurposeCodec
+      case 20 => changeSquadZoneCodec
+      case 21 => closeSquadMemberPositionCodec
+      case 22 => addSquadMemberPositionCodec
+      case 23 => changeSquadMemberRequirementsRoleCodec
+      case 24 => changeSquadMemberRequirementsDetailedOrdersCodec
+      case 25 => changeSquadMemberRequirementsCertificationsCodec
+      case 26 => resetAllCodec
+      case 28 => autoApproveInvitationRequestsCodec
+      case 31 => locationFollowsSquadLeadCodec
+      case 34 => searchForSquadsWithParticularRoleCodec
+      case 35 => cancelSquadSearchCodec
+      case 38 => assignSquadMemberToRoleCodec
+      case 39 => noSquadSearchResultsCodec
+      case 40 => findLfsSoldiersForRoleCodec
+      case 41 => cancelFindCodec
+      case  2 |  6 | 11 | 12 | 13 |
```

Comment:
Mainly I don't know what to call some of them or what exactly they do.  All of them are encountered, in the packets, infrequently and others more frequently.  Most of the UI discoveries were subsequently named.

`Unknown(6)`, for example, is used at the start of the client's game, that, is, before loading squad definitions loadouts, but I don't know what to call it.
`Unknown(18)` is called at the end of many squad operations, making it an unconfirmed "completion packet," but I don't know what to call it.
The majority are called out entirely for the benefit of the packet converter, and the failure condition is intended to indicate (for me) that I've encountered something new.

Reply:
Understood, my comment was mostly about duplicating the `unknownCodec(code)` inline with the rest of the cases just to avoid the large `N | N | N ... | N` syntax

Reply:
The cases have been broken up.